### PR TITLE
Fixes odin timeout not giving people their gear

### LINKED ISSUE #2580: [NEW MAP] Arrivals controller doesn't give you your items if you time out
When the timer count exceeds, you get shoved onto the Aurora but apparently missing your gear.

Tested job: surgeon.

### PR DESCRIPTION
**Pull request by printer16**
Odin timeout now gives people their gear appropriate to their rank.
Fixes #2580

### CODE REVIEW

File: code/controllers/subsystems/job.dm [hunk 1]
```diff
@@ -438,6 +438,9 @@
 			src << "<span class='warning'>You come to the sudden realization that you never left the Aurora at all! You were in cryo the whole time!</span>"
 			src.forceMove(pick(spawnpos.turfs))
 			global_announcer.autosay("[real_name], [mind.role_alt_title], [spawnpos.msg].", "Cryogenic Oversight")
+			if(!src.megavend)
```
Comment (skull132) [inline code/controllers/subsystems/job.dm]: Set `src.megavend = TRUE` after this as well. Just in case.

**Comment by Lohikar** [comment]
While you're touching the odin timeout, it seems that borgs do not have a functioning timeout.

Maybe move `odin_spawn_timeout` up to `mob/living`, overload it on humans, and move the timer set to somewhere reached by both borgs and crew?

**Comment by printer16** [comment]
DNM for now. Going to do more with it

File: code/controllers/subsystems/job.dm [hunk 2]
```diff
@@ -425,32 +425,49 @@
...
 
-/mob/living/carbon/human/proc/odin_timeout()
+/mob/living/proc/odin_timeout()
 	if (!istype(get_area(src), /area/centcom/spawning))
 		return
 
 	if (!client)
 		SSjobs.odin_despawn_mob(src)
-	else
+	else if(ishuman(src))
```
Comment (skull132) [inline code/controllers/subsystems/job.dm]: Hnrg.

`istype(src)` is bad and usually demonstrates bad objective oriented programming. Ideally, you wouldn't write code with those checks present. This is how to refactor it:
```
/mob/living/proc/odin_timeout()
	if (!istype(get_area(src), /area/centcom/spawning))
		return FALSE

	if (!client)
		SSjobs.odin_despawn_mob(src)
		return FALSE

	return TRUE

/mob/living/carbon/human/odin_timeout()
	. = ..()

	if (!.)
		return FALSE

	var/datum/spawnpoint/spawnpos = spawntypes["Cryogenic Storage"]
	if(spawnpos && istype(spawnpos))
		src << "<span class='warning'>You come to the sudden realization that you never left the Aurora at all! You were in cryo the whole time!</span>"
		src.forceMove(pick(spawnpos.turfs))
		global_announcer.autosay("[real_name], [mind.role_alt_title], [spawnpos.msg].", "Cryogenic Oversight")
		if(!src.megavend)
			var/rank = src.mind.assigned_role
			SSjobs.EquipRank(src, rank, 1)
			src.megavend = TRUE
	else
		SSjobs.odin_despawn_mob(src) //somehow they can't spawn at cryo, so this is the only recourse of action.

	return TRUE

/mob/living/silicon/robot/odin_timeout()
	. = ..()

	if (!.)
		return FALSE

	var/datum/spawnpoint/spawnpos = spawntypes["Cyborg Storage"]
	if(spawnpos && istype(spawnpos))
		src << "<span class='warning'>You come to the sudden realization that you never left the Aurora at all! You were in robotic storage the whole time!</span>"
		src.forceMove(pick(spawnpos.turfs))
		global_announcer.autosay("[real_name], [mind.role_alt_title], [spawnpos.msg].", "Robotic Oversight")
	else
		SSjobs.odin_despawn_mob(src)
```

The reason for this is cleanliness and extendability. It makes special cases much more logical to handle, and fall in line with how OOP is usually handled anyways.

**Comment by printer16** [comment]
Updated.

**Pull request merged**